aiden-recommender/aiden_recommender/scrapers/abstract_parser.py

```diff
@@ -0,0 +1,15 @@
+from aiden_recommender.models import JobOffer
+from aiden_recommender.scrapers.field_extractors import AbstractFieldExtractor
+from abc import ABC
+
+
+class AbstractParser(ABC):
+    def transform_to_job_offer(self, data) -> JobOffer:
+        for field in dir(self):
+            extractor = getattr(self, field)
+            if isinstance(extractor, AbstractFieldExtractor):
+                data = extractor.extract(data, field)
```

Comment:
Your input is a dictionary and the output is a dictionary, why not get the key and add it yourself ?

Reply:
At least you are not copying the dict each time since you are passing it by reference.
<img width="602" alt="image" src="https://github.com/Anko59/aiden-job-finder/assets/28636688/f5f02d7b-33a0-47b3-844b-c333a172a5a9">